omniledger/service/leader_map.go

```diff
@@ -0,0 +1,51 @@
+package service
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/dedis/onet/network"
+)
+
+type leaderMap struct {
```

Comment:
I don't understand why you need this `leaderMap` - can't you take the roster of the skipchain and look at the first entry?